osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -340,6 +341,8 @@ public override void SetReplayScore(Score replayScore)
         public void Attach(KeyCounterDisplay keyCounter) =>
             (KeyBindingInputManager as ICanAttachKeyCounter)?.Attach(keyCounter);
 
+        public void Attach(ClicksPerSecondCalculator calculator) => (KeyBindingInputManager as ICanAttachKeyCounter)?.Attach(calculator);
```

Comment:
Should not exist.

The whole flow should use a *single interface specification* which is shared between the two components. If you're not sure how this should work, I'll just implement it as it will be faster than attempting to explain.

Reply:
@ItsShamed shall i apply the changes required for this?

Reply:
Yes, please do so. Sadly I've not quite grasped what you said.